midend/eliminateActionRun.h

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2024 Intel Corp.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef MIDEND_ELIMINATEACTIONRUN_H_
+#define MIDEND_ELIMINATEACTIONRUN_H_
+
+#include "frontends/common/resolveReferences/resolveReferences.h"
+#include "ir/ir.h"
+
+namespace P4 {
+
+class ElimActionRun : public PassManager {
```

Comment:
Possible to add a comment with an example pre and post transformation here?

Reply:
done